src/lib/common/sol-platform-impl-systemd.c

```diff
@@ -489,3 +489,84 @@ sol_platform_impl_shutdown(void)
 
     sol_bus_close();
 }
+
+int
+sol_plataform_impl_set_hostname(const char *name)
+{
+    sd_bus *bus;
+    int r;
+
+    bus = sol_bus_get(NULL);
+    SOL_NULL_CHECK(bus, -ENOTCONN);
+
+    r = sd_bus_call_method_async(bus, NULL, "org.freedesktop.hostname1",
+        "/org/freedesktop/hostname1", "org.freedesktop.hostname1",
+        "SetStaticHostname", sol_bus_log_callback, NULL, "sb", name, false);
+    SOL_INT_CHECK(r, < 0, r);
+    return 0;
+}
+
+int
+sol_plataform_impl_get_hostname(char **name)
+{
+    sd_bus *bus;
+    sd_bus_error error = SD_BUS_ERROR_NULL;
+    int r;
+
+    bus = sol_bus_get(NULL);
+    SOL_NULL_CHECK(bus, -ENOTCONN);
+
+    r = sd_bus_get_property_string(bus, "org.freedesktop.hostname1",
```

Comment:
as I told you, it's simpler to just use `gethostname()` here, it's faster and you can't do it synchronously.

Usually set is the bitch, due permissions and so on. the daemon will do `sethosname()`, thus `gethostname()` will work